### PR TITLE
feat(client): reduce the main bundle size

### PR DESCRIPTION
**Pull request by leafty**
Fixes https://github.com/SwissDataScienceCenter/renku-ui/issues/2346.

Details:
* add bundle analyzer to the build
* use lazy loading to split the main bundle into smaller chunks

/deploy renku=auto-update/renku-ui-3.13.0 extra-values=notebooks.userSessionPersistentVolumes.enabled=true

# Before this PR

```
File sizes after gzip:

  3.33 MB    build/main.64ee47b01494ac98219b-dev.js
  343.95 kB  build/cde265b9865f3e759bd45f752b573529.js
  59.33 kB   build/static/css/main.fe878b3d.css
```

# After this PR

```
File sizes after gzip:

  970.56 kB  build/46.5b18a203683d9fa09704-dev.chunk.js
  951.45 kB  build/main.5b18a203683d9fa09704-dev.js
  722.23 kB  build/718.5b18a203683d9fa09704-dev.chunk.js
  343.95 kB  build/cde265b9865f3e759bd45f752b573529.js
  308.73 kB  build/535.5b18a203683d9fa09704-dev.chunk.js
  276.71 kB  build/901.5b18a203683d9fa09704-dev.chunk.js
  115.52 kB  build/73.5b18a203683d9fa09704-dev.chunk.js
  59.32 kB   build/static/css/main.95fd74aa.css
  1.33 kB    build/297.5b18a203683d9fa09704-dev.chunk.js
  599 B      build/575.5b18a203683d9fa09704-dev.chunk.js
  547 B      build/926.5b18a203683d9fa09704-dev.chunk.js
  494 B      build/378.5b18a203683d9fa09704-dev.chunk.js
  486 B      build/742.5b18a203683d9fa09704-dev.chunk.js
```

| Size | File | Component(s) | Package(s) |
|---|---|---|---|
| 970.56 kB | build/46.5b18a203683d9fa09704-dev.chunk.js | NotebookRender | react-ipynb-renderer |
| 722.23 kB | build/718.5b18a203683d9fa09704-dev.chunk.js | RenkuMarkdown, MarkdownTextExcerpt | mermaid, showdown, showdown-highlight, showdown-katex, he.js |
| 308.73 kB | build/535.5b18a203683d9fa09704-dev.chunk.js | CodePreview | highlight |
| 276.71 kB | build/901.5b18a203683d9fa09704-dev.chunk.js | CKEditor | @renku/ckeditor5-build-renku, @ckeditor/ckeditor5-react |
| 115.52 kB | build/73.5b18a203683d9fa09704-dev.chunk.js | PDFViewer | react-pdf |

# Lighthouse reports

Before:
image

After:
image

### CODE REVIEW

**Comment by lorenzo-cavazzi** [comment]
This fixes https://github.com/SwissDataScienceCenter/renku-ui/issues/2346

File: README.md [hunk 1]
```diff
@@ -179,6 +179,17 @@ We use [CSS modules](https://github.com/css-modules/css-modules) to apply CSS st
...
+
+If a component requires a large package, it can be loaded on demand by using the `lazy()` function from React.
+
+Here is an example:
+
+- The component itself: [NotebookRender.tsx](./client/src/components/notebook/NotebookRender.tsx)
+- The wrapper allowing for lazy loading: [LazyNotebookRender.tsx](./client/src/components/notebook/LazyNotebookRender.tsx)
+
+In this case, we save ~950kB from being included in the final bundle.
+
```
Comment (lorenzo-cavazzi) [inline README.md:192]: Should this go to the coding guidelines?

**Review by lorenzo-cavazzi — CHANGES_REQUESTED** [review]
Great improvement! :raised_hands: 

Some tests are failing. Feel free to remove any legacy Jest test trying to load a component in the DOM. It would be best if the same elements are tested at least once with Cypress or in Storybook

File: README.md (hunk 1 above)
Comment (leafty) [reply README.md:192]: Not sure, I prefer to have this here until I can write a good rule.

**Comment by leafty** [comment]
> Some tests are failing. Feel free to remove any legacy Jest test trying to load a component in the DOM. It would be best if the same elements are tested at least once with Cypress or in Storybook

- I have no idea why this test is failing. It runs very quickly on my computer.
- Tests like `display a project` render and check some of the touched components (e.g. notebook render, code render)

**Comment by leafty** [comment]
Note about the removed tests: it seems that jest tests are not working in CI/CD with <Suspense>. The corresponding components should be covered by Cypress tests.

**Review by lorenzo-cavazzi — APPROVED** [review]

**Pull request merged**